File: samples/server/petstore/sinatra/api/pet_api.rb
require 'json'


<<<<<<< HEAD
MyApp.add_route('POST', '/v2/pets', {
=======
MyApp.add_route('POST', '/v2/pet', {
>>>>>>> upstream/master
  "resourcePath" => "/Pet",
  "summary" => "Add a new pet to the store",
  "nickname" => "add_pet", 
  "responseClass" => "void", 
  "endpoint" => "/pets", 
  "notes" => "",
  "parameters" => [
    {
      "name" => "body",
      "description" => "Pet object that needs to be added to the store",
      "dataType" => "Pet",
      "paramType" => "body",
    }
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


<<<<<<< HEAD
MyApp.add_route('DELETE', '/v2/pets/{petId}', {
=======
MyApp.add_route('DELETE', '/v2/pet/{petId}', {
>>>>>>> upstream/master
  "resourcePath" => "/Pet",
  "summary" => "Deletes a pet",
  "nickname" => "delete_pet", 
  "responseClass" => "void", 
<<<<<<< HEAD
  "endpoint" => "/pets/{petId}", 
  "notes" => "",
  "parameters" => [
    
    
    {
      "name" => "pet_id",
      "description" => "Pet id to delete",
      "dataType" => "int",
      "paramType" => "path",
    },
    
    
    {
=======
  "endpoint" => "/pet/{petId}", 
  "notes" => "",
  "parameters" => [
    {
      "name" => "pet_id",
      "description" => "Pet id to delete",
      "dataType" => "int",
      "paramType" => "path",
    },
    {
>>>>>>> upstream/master
      "name" => "api_key",
      "description" => "",
      "dataType" => "string",
      "paramType" => "header",
    },
<<<<<<< HEAD
    
    
=======
>>>>>>> upstream/master
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/v2/pets/findByStatus', {
  "resourcePath" => "/Pet",
  "summary" => "Finds Pets by status",
  "nickname" => "find_pets_by_status", 
  "responseClass" => "array[Pet]", 
<<<<<<< HEAD
  "endpoint" => "/pets/findByStatus", 
  "notes" => "Multiple status values can be provided with comma seperated strings",
=======
  "endpoint" => "/pet/findByStatus", 
  "notes" => "Multiple status values can be provided with comma separated strings",
>>>>>>> upstream/master
  "parameters" => [
    {
      "name" => "status",
      "description" => "Status values that need to be considered for filter",
      "dataType" => "array[string]",
      "paramType" => "query",
      "collectionFormat" => "csv",
      "allowableValues" => "",
      
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/v2/pets/findByTags', {
  "resourcePath" => "/Pet",
  "summary" => "Finds Pets by tags",
  "nickname" => "find_pets_by_tags", 
  "responseClass" => "array[Pet]", 
<<<<<<< HEAD
  "endpoint" => "/pets/findByTags", 
  "notes" => "Muliple tags can be provided with comma seperated strings. Use tag1, tag2, tag3 for testing.",
=======
  "endpoint" => "/pet/findByTags", 
  "notes" => "Multiple tags can be provided with comma separated strings. Use tag1, tag2, tag3 for testing.",
>>>>>>> upstream/master
  "parameters" => [
    {
      "name" => "tags",
      "description" => "Tags to filter by",
      "dataType" => "array[string]",
      "paramType" => "query",
      "collectionFormat" => "csv",
      "allowableValues" => "",
      
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/v2/pets/{petId}', {
  "resourcePath" => "/Pet",
  "summary" => "Find pet by ID",
  "nickname" => "get_pet_by_id", 
  "responseClass" => "Pet", 
<<<<<<< HEAD
  "endpoint" => "/pets/{petId}", 
  "notes" => "Returns a pet when ID < 10.  ID > 10 or nonintegers will simulate API error conditions",
=======
  "endpoint" => "/pet/{petId}", 
  "notes" => "Returns a single pet",
>>>>>>> upstream/master
  "parameters" => [
    {
      "name" => "pet_id",
      "description" => "ID of pet to return",
      "dataType" => "int",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


<<<<<<< HEAD
MyApp.add_route('PUT', '/v2/pets', {
  "resourcePath" => "/Pet",
  "summary" => "Update an existing pet",
  "nickname" => "update_pet", 
=======
MyApp.add_route('PUT', '/v2/pet', {
  "resourcePath" => "/Pet",
  "summary" => "Update an existing pet",
  "nickname" => "update_pet", 
  "responseClass" => "void", 
  "endpoint" => "/pet", 
  "notes" => "",
  "parameters" => [
    {
      "name" => "body",
      "description" => "Pet object that needs to be added to the store",
      "dataType" => "Pet",
      "paramType" => "body",
    }
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('POST', '/v2/pet/{petId}', {
  "resourcePath" => "/Pet",
  "summary" => "Updates a pet in the store with form data",
  "nickname" => "update_pet_with_form", 
>>>>>>> upstream/master
  "responseClass" => "void", 
  "endpoint" => "/pets", 
  "notes" => "",
  "parameters" => [
<<<<<<< HEAD
    
    
    
    
    {
      "name" => "body",
      "description" => "Pet object that needs to be added to the store",
      "dataType" => "Pet",
      "paramType" => "body",
    }
    
=======
    {
      "name" => "pet_id",
      "description" => "ID of pet that needs to be updated",
      "dataType" => "int",
      "paramType" => "path",
    },
>>>>>>> upstream/master
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('POST', '/v2/pets/{petId}', {
  "resourcePath" => "/Pet",
<<<<<<< HEAD
  "summary" => "Updates a pet in the store with form data",
  "nickname" => "update_pet_with_form", 
  "responseClass" => "void", 
  "endpoint" => "/pets/{petId}", 
=======
  "summary" => "uploads an image",
  "nickname" => "upload_file", 
  "responseClass" => "ApiResponse", 
  "endpoint" => "/pet/{petId}/uploadImage", 
>>>>>>> upstream/master
  "notes" => "",
  "parameters" => [
    {
      "name" => "pet_id",
      "description" => "ID of pet that needs to be updated",
      "dataType" => "string",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end

File: samples/client/petstore/java/retrofit2/src/main/java/io/swagger/client/api/PetApi.java
package io.swagger.client.api;

import io.swagger.client.CollectionFormats.*;




import retrofit2.Call;

import retrofit2.http.*;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;

import io.swagger.client.model.ModelApiResponse;
import io.swagger.client.model.Pet;



import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;



public interface PetApi {
  
  /**
   * Add a new pet to the store
   * 

   * @param pet Pet object that needs to be added to the store (required)

   * @return Call&lt;Void&gt;

   */
  
  
  
  
  @Headers({
    "Content-Type:application/json"
  })
  
  
  
  
    
  @POST("pet")
  Call<Void> addPet(
    @retrofit2.http.Body Pet pet
  );

  
  /**
   * Deletes a pet
   * 

   * @param petId Pet id to delete (required)

   * @param apiKey  (optional)

   * @return Call&lt;Void&gt;

   */
  
  
  
    
  @DELETE("pet/{petId}")
  Call<Void> deletePet(
    @retrofit2.http.Path("petId") Integer petId, @retrofit2.http.Header("api_key") String apiKey
  );

  
  /**
   * Finds Pets by status
   * Multiple status values can be provided with comma separated strings

   * @param status Status values that need to be considered for filter (required)

   * @return Call&lt;List&lt;Pet&gt;&gt;

   */
  
  
  
    
  @GET("pet/findByStatus")
  Call<List<Pet>> findPetsByStatus(
    @retrofit2.http.Path("status") List<String> status
  );

  
  /**
   * Finds Pets by tags
   * Multiple tags can be provided with comma separated strings. Use tag1, tag2, tag3 for testing.

   * @param tags Tags to filter by (required)

   * @return Call&lt;List&lt;Pet&gt;&gt;

   */
  
  
  
    
  @GET("pet/findByTags")
  Call<List<Pet>> findPetsByTags(
    @retrofit2.http.Path("tags") List<String> tags
  );

  
  /**
   * Find pet by ID
   * Returns a single pet

   * @param petId ID of pet to return (required)

   * @return Call&lt;Pet&gt;

   */
  
  
  
    
  @GET("pet/{petId}")
  Call<Pet> getPetById(
    @retrofit2.http.Path("petId") Integer petId
  );

  
  /**
   * Update an existing pet
   * 

   * @param pet Pet object that needs to be added to the store (required)

   * @return Call&lt;Void&gt;

   */
  
  
  
  
  @Headers({
    "Content-Type:application/json"
  })
  
  
  
  
    
  @PUT("pet")
  Call<Void> updatePet(
    @retrofit2.http.Body Pet pet
  );

  
  /**
   * Updates a pet in the store with form data
   * 

   * @param petId ID of pet that needs to be updated (required)

   * @param body  (optional)

   * @return Call&lt;Void&gt;

   */
  
  
  
  
  @Headers({
    "Content-Type:application/x-www-form-urlencoded"
  })
  
  
    
  @POST("pet/{petId}")
  Call<Void> updatePetWithForm(
    @retrofit2.http.Path("petId") Integer petId, @retrofit2.http.Body Object body
  );

  
  /**
   * uploads an image
   * 

   * @param petId ID of pet to update (required)

   * @param body  (optional)

   * @return Call&lt;ModelApiResponse&gt;

   */
  
  
  
    
  @POST("pet/{petId}/uploadImage")
  Call<ModelApiResponse> uploadFile(
    @retrofit2.http.Path("petId") Integer petId, @retrofit2.http.Body Object body
  );

  
}

